From ca039d94f162f858d3d4dae29ecaadeecd20e1a2 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:58:49 -0400
Subject: [fs] btrfs: fix extent boundary check in bio_readpage_error

Message-id: <1407179109-13714-43-git-send-email-sandeen@redhat.com>
Patchwork-id: 86988
O-Subject: [RHEL7.1 PATCH 042/422] Btrfs: fix extent boundary check in bio_readpage_error
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Filipe David Borba Manana <fdmanana@gmail.com>

commit 68ba990f7d161c31e9eddd98727ba8393089047f
Author: Filipe David Borba Manana <fdmanana@gmail.com>
Date:   Mon Nov 25 03:22:07 2013 +0000

    Btrfs: fix extent boundary check in bio_readpage_error

    Signed-off-by: Filipe David Borba Manana <fdmanana@gmail.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/extent_io.c b/fs/btrfs/extent_io.c
index 9f580f2..3ec5f05 100644
--- a/fs/btrfs/extent_io.c
+++ b/fs/btrfs/extent_io.c
@@ -2156,7 +2156,7 @@ static int bio_readpage_error(struct bio *failed_bio, u64 phy_offset,
    return -EIO;
   }
 
-  if (em->start > start || em->start + em->len < start) {
+  if (em->start > start || em->start + em->len <= start) {
    free_extent_map(em);
    em = NULL;
   }
-- 
1.7.1